services/provider/server/server.go

```diff
@@ -320,104 +324,465 @@ func newScheme() (*runtime.Scheme, error) {
+		}
 	}
-	err = s.client.List(ctx, podList, listOpts...)
-	if err != nil {
-		return nil, fmt.Errorf("failed to list pod with rook-ceph-mgr label. %v", err)
+	cephFsClientProfile.CephFs = &csiopv1a1.CephFsConfigSpec{
+		SubVolumeGroup:     consumerConfig.GetSubVolumeGroupName(),
+		KernelMountOptions: kernelMountOptions,
+		RadosNamespace:     ptr.To(consumerConfig.GetSubVolumeGroupRadosNamespaceName()),
 	}
-	if len(podList.Items) == 0 {
-		return nil, fmt.Errorf("no pods available with rook-ceph-mgr label")
+	profileMap[cephFsClientProfileName] = cephFsClientProfile
+
+	nfsClientProfileName := consumerConfig.GetCephFsClientProfileName()
+	nfsClientProfile := profileMap[nfsClientProfileName]
+	if nfsClientProfile == nil {
+		nfsClientProfile = &csiopv1a1.ClientProfileSpec{
+			CephConnectionRef: corev1.LocalObjectReference{Name: consumer.Status.Client.Name},
+		}
 	}
+	nfsClientProfile.Nfs = &csiopv1a1.NfsConfigSpec{}
+	profileMap[nfsClientProfileName] = nfsClientProfile
 
-	mgrPod := &podList.Items[0]
-	var port int32 = -1
+	for profileName, profileSpec := range profileMap {
+		extR = append(extR, &pb.ExternalResource{
+			Kind: "ClientProfile",
+			Name: profileName,
+			Data: mustMarshal(profileSpec),
+		})
+	}
+	return extR, nil
+}
 
-	for i := range mgrPod.Spec.Containers {
-		container := &mgrPod.Spec.Containers[i]
-		if container.Name == "mgr" {
-			for j := range container.Ports {
-				if container.Ports[j].Name == "http-metrics" {
-					port = container.Ports[j].ContainerPort
-				}
-			}
+func (s *OCSProviderServer) appendCephClientSecretExternalResources(
+	ctx context.Context,
+	extR []*pb.ExternalResource,
+	consumer *ocsv1alpha1.StorageConsumer,
+	consumerConfig util.StorageConsumerResources,
+) ([]*pb.ExternalResource, error) {
+
+	cephClients := []string{
+		consumerConfig.GetCsiRbdProvisionerSecretName(),
+		consumerConfig.GetCsiRbdNodeSecretName(),
+		consumerConfig.GetCsiCephFsProvisionerSecretName(),
+		consumerConfig.GetCsiCephFsNodeSecretName(),
+	}
+
+	for i := range cephClients {
+		cephClient := &rookCephv1.CephClient{}
+		cephClient.Name = cephClients[i]
+		cephClient.Namespace = consumer.Namespace
+		if err := s.client.Get(ctx, client.ObjectKeyFromObject(cephClient), cephClient); err != nil {
+			return extR, err
+		}
+
+		cephUserSecret := &v1.Secret{}
+		cephUserSecret.Namespace = consumer.Namespace
+		if cephClient.Status != nil &&
+			cephClient.Status.Info["secretName"] != "" {
+			cephUserSecret.Name = cephClient.Status.Info["secretName"]
+		}
+		if cephUserSecret.Name == "" {
+			return extR, fmt.Errorf("failed to find cephclient secret name")
+		}
+
+		if err := s.client.Get(ctx, client.ObjectKeyFromObject(cephUserSecret), cephUserSecret); err != nil {
+			return extR, fmt.Errorf("failed to get %s secret. %v", cephUserSecret, err)
 		}
+
+		secretContent := map[string]string{}
+		for key, value := range cephUserSecret.Data {
+			secretContent[key] = string(value)
+		}
```

Comment:
We should be sending `map[string]bytes` and not  `map[string]string`. We should fix the receiver side to handle `map[string]byte`
```suggestion
```